### PR TITLE
[Plugin] Implement custom window tabs

### PR DESCRIPTION
**Pull request by IntelOrca**
* Implement tabs on custom windows.
* Fix off-by-one error with control sizes.
* Improve window resize support.
* Add border property to button so image buttons can optionally have borders.
* Add `onUpdate` event to window.
* Fix UTF-8 strings not showing correctly.
* Fix re-positioning of widgets on a window.
* Add support for tooltips.

### CODE REVIEW

File: src/openrct2-ui/scripting/CustomWindow.cpp [hunk 1]
```diff
@@ -172,6 +175,43 @@ namespace OpenRCT2::Ui::Windows
...
+        }
+    };
+
+    struct CustomTabDesc
+    {
+        ImageId imageFrameBase;
+        uint32_t imageFrameCount;
+        uint32_t imageFrameDuration;
+        int32_t offsetX;
+        int32_t offsetY;
```
Comment (tupaschoal) [inline src/openrct2-ui/scripting/CustomWindow.cpp:189]: Can't (and shouldn't we) use `ScreenCoordsXY` for this?

File: src/openrct2-ui/scripting/CustomWindow.cpp [hunk 2]
```diff
@@ -202,8 +201,13 @@ namespace OpenRCT2::Ui::Windows
...
                 result.imageFrameCount = AsOrDefault(dukImage["frameCount"], 0);
                 result.imageFrameDuration = AsOrDefault(dukImage["frameDuration"], 0);
-                result.offsetX = AsOrDefault(dukImage["offsetX"], 0);
-                result.offsetY = AsOrDefault(dukImage["offsetY"], 0);
+
+                auto dukCoord = dukImage["offset"];
+                if (dukCoord.type() == DukValue::Type::OBJECT)
+                {
+                    result.offset.x = AsOrDefault(dukCoord["x"], 0);
+                    result.offset.y = AsOrDefault(dukCoord["y"], 0);
```
Comment (tupaschoal) [inline src/openrct2-ui/scripting/CustomWindow.cpp:209]: Alternatively:
```suggestion
                    result.offset = { AsOrDefault(dukCoord["x"], 0), AsOrDefault(dukCoord["y"], 0) };
```

File: distribution/openrct2.d.ts [hunk 3]
```diff
@@ -822,8 +822,7 @@ declare global {
         frameBase: number;
         frameCount?: number;
         frameDuration?: number;
-        offsetX?: number;
-        offsetY?: number;
+        offset?: Coord2;
```
Comment (tupaschoal) [inline distribution/openrct2.d.ts:825]: We should rename `Coord2` and `Coord3` to either the same as their counterparts in C++ or something more meaningful at some point

Comment (IntelOrca) [reply distribution/openrct2.d.ts:825]: At the moment they are used for both map and screen coordinates. There is no strict type checking in JavaScript and TypeScript will be happy if the interface has the same fields which they will do.

Maybe I will change it, but in a separate PR.

Comment (tupaschoal) [reply distribution/openrct2.d.ts:825]: Yeah, I'm ok with it being in a different PR (I thought it would be), just don't want it to be forgotten.

**Pull request merged**